torchinductor/lowering.py

```diff
@@ -460,19 +472,32 @@ def _embedding_bag(
     )
 
 
-"""
 @register_lowering(aten._cudnn_rnn, type_promote=False)
 def _cudnn_rnn(*args):
     return list(
         map(
             TensorBox.create,
-            ir.FallbackKernel.create(
-                aten._cudnn_rnn,
-                *args
-            ),
+            ir.FallbackKernel.create(aten._cudnn_rnn, *args),
         )
     )
-"""
+
+
+@register_lowering(aten.grid_sampler_2d, type_promote=False)
```

Comment:
The decompositions in core are aiming to accurately capture type promotion logic (although certainly possible that they're not completely correct in some cases).

Reply:
Does this already exist in core?  I didn't see it when I wrote this (last week).

Happy to switch when it is available.

Reply:
ah, I was specifically asking about the `type_promote` flag. `grid_sampler_2d` is not in core.